server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/InternalHistogram.java

```diff
@@ -399,8 +399,8 @@ public void accept(InternalAggregation aggregation) {
             @Override
             public InternalAggregation get() {
                 List<Bucket> reducedBuckets = reducer.get();
+                reducedBuckets.sort(Comparator.comparingDouble(b -> b.key));
                 if (reduceContext.isFinalReduce()) {
-                    reducedBuckets.sort(Comparator.comparingDouble(b -> b.key));
```

Comment:
We only need to do this if we're talking to an older version, right? I'm fine doing it all the time just to fix the bug though. I guess we could sort on the way out of the wire? Meh. This is fine.